model/src/main/java/com/redhat/ipaas/model/integration/Integration.java

```diff
@@ -31,9 +31,7 @@
 @JsonDeserialize(builder = Integration.Builder.class)
 public interface Integration extends WithId<Integration>, WithName, Serializable {
 
-    public static enum Type {Activated, Deactivated, Draft};
-
-    public static enum Phase {Pending, Running, Succeeded, Failed, Unknown};
+    public static enum Status { Draft, Activated, Deactivated, Pending, Deleted};
```

Comment:
I don't think you ever desire for the state to be pending. I'd prefer to keep them separated.